spec/liquid_tags/github_tag/github_code_tag_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "rails_helper"
+
+vcr_option = {
+  cassette_name: "github_api_code",
+  allow_playback_repeats: "true"
+}
+
+RSpec.describe GithubTag::GithubCodeTag, vcr: vcr_option do
+  describe "#id" do
+    let(:file_name) { "johnpapa/vscode-peacock/.vscodeignore" }
+    let(:path) { "https://github.com/johnpapa/vscode-peacock/blob/master/.vscodeignore" }
+
+    setup { Liquid::Template.register_tag("github", GithubTag) }
+
+    def generate_github_code(path, line_number, _options = "")
+      Liquid::Template.parse("{% github #{path}#{line_number} %}")
+    end
+
+    # need to figure out how to generate a token that works
```

Comment:
you can configure your local dev installation with github, see https://docs.dev.to/backend/auth-github/ and then when you'll have `GITHUB_KEY` and `GITHUB_SECRET` configured in your `application.yml` they will be picked up by the `Github::Client` class.

Then you can use VCR. Don't worry, VCR won't store your secrets in the yml file (double check :)).

Example of you to use `VCR`: 

https://github.com/thepracticaldev/dev.to/blob/3b188d349ad018d41c88ee2a9e25ca1dedac06a3/spec/liquid_tags/github_tag/github_readme_tag_spec.rb#L36-L41

don't forget to enable VCR for the file:

https://github.com/thepracticaldev/dev.to/blob/3b188d349ad018d41c88ee2a9e25ca1dedac06a3/spec/liquid_tags/github_tag/github_readme_tag_spec.rb#L3